### PR TITLE
INV4 pallet add comments: partial completion

### PR DESCRIPTION
**Pull request by jakerumbles**
Added helpful comments throughout the INV4 pallet except for part of ipt.rs and ipl.rs.

#170 
#171 
#172
#173
#174

### CODE REVIEW

File: INV4/pallet-inv4/src/ips.rs [hunk 1]
```diff
@@ -613,12 +658,14 @@ impl<T: Config> Pallet<T> {
                 default_permission: ipl_default_permission,
             };
 
+            // ???
```
Comment (arrudagates) [inline INV4/pallet-inv4/src/ips.rs:661]: This is minting existential deposit of IPT0 to the creator of the replica, don't worry about commenting this as it's obsolete and needs to be rewritten.

File: INV4/pallet-inv4/src/ipt.rs [hunk 2]
```diff
@@ -56,29 +57,34 @@ impl<T: Config> Pallet<T> {
     ) -> DispatchResult {
         let owner = ensure_signed(owner)?;
 
+        // IP Set must exist for their to be a token
```
Comment (arrudagates) [inline INV4/pallet-inv4/src/ipt.rs:60]:
```suggestion
        // IP Set must exist for there to be a token
```

File: INV4/pallet-inv4/src/lib.rs [hunk 3]
```diff
@@ -233,19 +248,28 @@ pub mod pallet {
...
+        /// IpInfo (IPS) struct updated: assets removed from IPS. Optionally, new metadata set
         Removed(T::AccountId, T::IpId, Vec<u8>, Vec<AnyIdWithNewOwner<T>>),
+        /// Replicas of this IP Set are now allowed
         AllowedReplica(T::IpId),
+        /// Replicas of this IP Set are no longer allowed
         DisallowedReplica(T::IpId),
+        /// A replica of this IP Set was created
         ReplicaCreated(T::AccountId, T::IpId, T::IpId),
 
+        /// Sub tokens were minted/IPF was created???
```
Comment (arrudagates) [inline INV4/pallet-inv4/src/lib.rs:266]: Tokens minted, you won't find IPF related events here, as those are defined in the IPF pallet.
```suggestion
        /// Sub tokens were minted
```

File: INV4/pallet-inv4/src/lib.rs [hunk 4]
```diff
@@ -233,19 +248,28 @@ pub mod pallet {
...
         Minted(
             (T::IpId, Option<T::IpId>),
             T::AccountId,
             <T as pallet::Config>::Balance,
         ),
+        /// Sub tokens were burned/IPF was deleted???
```
Comment (arrudagates) [inline INV4/pallet-inv4/src/lib.rs:272]: Tokens burned, you won't find IPF related events here, as those are defined in the IPF pallet.
```suggestion
        /// Sub tokens were burned
```

File: primitives/src/lib.rs [hunk 5]
```diff
@@ -32,19 +37,20 @@ pub enum BoolOrWasm<Wasm> {
...
     pub metadata: IpsMetadataOf,
-    /// IPS Properties
+    /// IPS children. Holds list of all items the IP Set directly owns.
     pub data: Data,
     /// IPS Type
     pub ips_type: IpsType<IpId>,
     /// If this IPS allows replicas
     pub allow_replica: bool,
-
+    /// Specifically, the supply of IPT0 (ownership) tokens???
```
Comment (arrudagates) [inline primitives/src/lib.rs:53]:
```suggestion
    /// Specifically, the supply of IPT0 (ownership) tokens.
```

File: INV4/pallet-inv4/src/ipt.rs (hunk 2 above)
Comment (jakerumbles) [reply INV4/pallet-inv4/src/ipt.rs:60]: Been trying to figure out these 2 words for 26 years

File: INV4/pallet-inv4/src/lib.rs (hunk 3 above)
Comment (jakerumbles) [reply INV4/pallet-inv4/src/lib.rs:266]: ah ok woops, event has the same name in ipf pallet

**Pull request merged**